pkg/typesystem/typesystem.go

```diff
@@ -331,6 +331,35 @@ func (t *TypeSystem) DirectlyRelatedUsersets(objectType, relation string) ([]*op
 	return usersetRelationReferences, nil
 }
 
+// ResolvesExclusivelyToDirectlyAssignable returns whether all relationReferences are relations that are exclusively directly assignable.
+// For now, it will return false if the directly assignable relations are public wildcard, contains condition, or is another userset because
+// check resolver using these relations cannot be evaluated via simple datastore query.
+func (t *TypeSystem) ResolvesExclusivelyToDirectlyAssignable(relationReferences []*openfgav1.RelationReference) (bool, error) {
+	for _, rr := range relationReferences {
+		if _, ok := rr.GetRelationOrWildcard().(*openfgav1.RelationReference_Relation); !ok {
+			return false, nil
+		}
+		relation, err := t.GetRelation(rr.GetType(), rr.GetRelation())
+		if err != nil {
+			return false, err
```

Comment:
I think we should eventually evolve the return signature of this function to just be bool

Reply:
we can do it in future improvement